server/test/modules/provisioning/infrastructureConfigurationProvider.spec.js

```diff
@@ -0,0 +1,174 @@
+/* TODO: enable linting and fix resulting errors */
+/* eslint-disable */
+/* Copyright (c) Trainline Limited, 2016-2017. All rights reserved. See LICENSE.txt in the project root for license information. */
+'use strict';
+
+let should = require("should");
+let sinon = require("sinon");
+let proxyquire = require('proxyquire');
+let DynamoItemNotFoundError = require('../../../modules/errors/DynamoItemNotFoundError.class');
+
+describe("infrastructureConfigurationProvider:", () => {
+
+  const input = {
+    environmentName: "st1",
+    serviceName: "AService",
+    serverRoleName: "ServerRoleName"
+  };
+
+  let serverRole, depsResults, deps;
+
+  function setupDependencyResults() {
+    serverRole = {
+      ServerRoleName: input.serverRoleName,
+      OwningCluster: "Ransom",
+      AMI: "TheAmi"
+    };
+
+    depsResults = {
+      environment: {
+        getDeploymentMap() {
+          return {
+            DeploymentTarget: [serverRole]
+          };
+        },
+        EnvironmentType: "EnvironmentType"
+      },
+      service: {},
+      environmentType: {},
+      cluster: {
+        ClusterName: "Ransom",
+        Value: {
+          ShortName: "rm",
+          KeyPair: "something"
+        }
+      },
+      image: {}
+    };
+  }
+
+  function setupDependencies() {
+    setupDependencyResults();
+    deps = {
+      environmentMock: {
+        getByName: sinon.stub().returns(Promise.resolve(depsResults.environment)),
+      },
+      servicesDbMock: {
+        get: sinon.stub().returns(Promise.resolve(depsResults.service))
+      },
+      environmentTypeMock: {
+        getByName: sinon.stub().returns(Promise.resolve(depsResults.environmentType))
+      },
+      clustersMock: {
+        get: sinon.stub().returns(Promise.resolve(depsResults.cluster))
+      },
+      imageProviderMock: {
+        get: sinon.stub().returns(Promise.resolve(depsResults.image))
+      }
+    };
+  }
+
+  function getInstance() {
+    var mapStubs = {};
+    mapStubs['./launchConfiguration/imageProvider'] = deps.imageProviderMock;
+    mapStubs['../../models/Environment'] = deps.environmentMock;
+    mapStubs['../../models/EnvironmentType'] = deps.environmentTypeMock;
+    mapStubs['../data-access/clusters'] = deps.clustersMock;
+    mapStubs['../data-access/services'] = deps.servicesDbMock;
+    var infrastructureConfigurationProvider = proxyquire('../../../modules/provisioning/infrastructureConfigurationProvider', mapStubs);
+
+    var target = infrastructureConfigurationProvider;
+
+    return target.get(input.environmentName, input.serviceName, input.serverRoleName);
+  }
```

Comment:
Could these helper functions get shifter to the bottom, so the file shows the tests first.

Reply:
👍